heclib/heclib_c/src/Public/zpdStore.c

```diff
@@ -274,9 +294,541 @@ int zpdStore(long long *ifltab, zStructPairedData *pds, int storageFlag)
+				//------------------------//
+				if (vdiPd->offsetToNgvd29 != vdiLoc->offsetToNgvd29) {
+					sprintf(
+						errmsg,
+						"\nIncoming offset to NGVD_29 of %f %s conflicts with existing offset of %f %s.\n"
+						"Call 'zset(\"VDOW\", \"\", 1)' to allow overwriting the location's vertical datum information.\n"
+						"Conversion to datum '%s' was not performed.\n"
+						"No data stored.",
+						vdiPd->offsetToNgvd29, vdiPd->unit, vdiLoc->offsetToNgvd29, vdiLoc->unit, vdiLoc->nativeDatum);
+					if (vdiPd != &_vdiPd) {
+						free(vdiPd);
+					}
+					return zerrorProcessing(ifltab, DSS_FUNCTION_zpdStore_ID,
+						zdssErrorCodes.INCOMPATIBLE_CALL, 0,
+						0, zdssErrorSeverity.WARNING, pds->pathname,
+						errmsg);
+				}
+				vdi = vdiLoc;
+			}
+		}
+		else {
+			//--------------------------//
+			// use the only one we have //
+			//--------------------------//
+			vdi = vdiPd ? vdiPd : vdiLoc;
+		}
+		if (vdi) {
+			//----------------------------------//
+			// get the effective vertical datum //
+			//----------------------------------//
+			double offset = 0.;
+			char   cvertical_datum[CVERTICAL_DATUM_SIZE];
+			int    ivertical_datum = -1;
+			int    headerCopyNumber = pds->userHeaderNumber;
+			int   *headerCopy = NULL;
+			if (headerCopyNumber > 0) {
+				headerCopy = (int *)calloc(headerCopyNumber, 4);
+				memcpy(headerCopy, pds->userHeader, headerCopyNumber * 4);
+			}
+			if (indElev) {
+				ivertical_datum = getEffectiveVerticalDatum(
+					cvertical_datum,
+					sizeof(cvertical_datum),
+					&pds->userHeader,
+					&pds->userHeaderNumber,
+					&pds->unitsIndependent);
+				//-------------------------------------------------------//
+				// now that we have a datum, determine the offset to use //
+				//-------------------------------------------------------//
+				switch(ivertical_datum) {
+					case IVERTICAL_DATUM_NAVD88 :
+						offset = vdi->offsetToNavd88;
+						break;
+					case IVERTICAL_DATUM_NGVD29 :
+						offset = vdi->offsetToNgvd29;
+						break;
+					default :
+						if(!strcmp(cvertical_datum, vdi->nativeDatum) || !strcmp(cvertical_datum, CVERTICAL_DATUM_OTHER)) {
+							offset = 0;
+						}
+						else {
+							offset = UNDEFINED_VERTICAL_DATUM_VALUE;
+						}
+						break;
+				}
+				if (offset != 0.) {
+					char errmsg[256];
+					if (offset == UNDEFINED_VERTICAL_DATUM_VALUE) {
+						sprintf(
+							errmsg,
+							"\nVertical datum offset is undefined for datum '%s'.\n"
+							"Datum conversion could not be performed.\n"
+							"No data stored.",
+							cvertical_datum);
+						if (vdiPd && vdiPd != &_vdiPd) {
+							free(vdiPd);
+						}
+						return zerrorProcessing(ifltab, DSS_FUNCTION_zpdStore_ID,
+							zdssErrorCodes.INCOMPATIBLE_CALL, 0,
+							0, zdssErrorSeverity.WARNING, pds->pathname,
+							errmsg);
+					}
+					offset = getOffset(offset, vdi->unit, indElev ? pds->unitsIndependent : pds->unitsDependent);
+					if (offset == UNDEFINED_VERTICAL_DATUM_VALUE) {
```

Comment:
The following if block is redundant - we are already in an if block where indElev == TRUE